### PR TITLE
Remove hal dependencies from the crate

### PR DESCRIPTION
**Pull request by Disasm**
This way `stm32-usbd` does not rely on `stm32*-hal` dependencies, instead hals use this driver for the corresponding peripherals.

See also: https://github.com/stm32-rs/stm32-usbd/issues/14

### CODE REVIEW

**Comment by Disasm** [comment]
This PR is tested on F042, L432, F303, F103.

**Comment by therealprof** [comment]
Sounds fantastic!

File: examples/hal.rs
Comment (therealprof) [inline examples/hal.rs]: Uhm, it would be great if the example actually showed how to do some USB interaction instead of just initialising something. 😅

Comment (Disasm) [reply examples/hal.rs]: This is a "library" example. I'm going to add actual USB examples to the corresponding hal libraries.

Comment (therealprof) [reply examples/hal.rs]: Cool, disregard this comment then. 😀

**Review by TeXitoi — APPROVED** [review]
Great!

**Review by mvirkkunen — APPROVED** [review]
LGTM as long as you can get the HAL maintainers on board with adding this!

**Pull request merged**
